src/main/scala/circt/stage/phases/CIRCT.scala

```diff
@@ -61,40 +63,48 @@ class CIRCT extends Phase {
     var blackbox, inferReadWrite = false
     var dedup, imcp = true
     var logLevel = _root_.logger.LogLevel.None
+    var split = false
 
-    val annotationsx = annotations.filter {
-      case _: CustomFileEmission           => false
-      case _: firrtl.EmitCircuitAnnotation => false
+    val annotationsx: AnnotationSeq = annotations.flatMap {
+      case a: CustomFileEmission => {
+        val filename = a.filename(annotations)
+        a.replacements(filename)
+      }
+      case _: firrtl.EmitCircuitAnnotation => Nil
+      case _: firrtl.EmitAllModulesAnnotation => {
+        split = true
+        Nil
+      }
       case RunFirrtlTransformAnnotation(transform) =>
         transform match {
           /* Inlining/Flattening happen by default, so these can be dropped. */
-          case _: firrtl.passes.InlineInstances | _: firrtl.transforms.Flatten => false
+          case _: firrtl.passes.InlineInstances | _: firrtl.transforms.Flatten => Nil
           /* ReplSeqMem is converted to a firtool option */
           case _: firrtl.passes.memlib.ReplSeqMem =>
             blackbox = true
-            false
+            Nil
           /* Any emitters should not be passed to firtool. */
-          case _: firrtl.Emitter => false
+          case _: firrtl.Emitter => Nil
           /* Default case: leave the annotation around and let firtool warn about it. */
-          case _ => true
+          case _ => Nil
```

Comment:
This is changing behavior. Previously, all non-special cased `RunFirrtlTransformAnnotation`s are kept around for CIRCT to error about them.
```suggestion
          case a => Seq(a)
```